### ISSUE TITLE
slcli server detail example01 (KeyError: 'tag')

### ISSUE THREAD

**Issue by mbubb** [softlayer/softlayer-python#570]
Hello 

I am running 

slcli server detail hostname

against our entire SLR inventory (200 servers) for most of them I get expected output but for 4 out of the 100 I get this:

```
➜  ~  slcli server detail example01
An unexpected error has occured:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/SoftLayer-4.0.2-py2.7.egg/SoftLayer/CLI/core.py", line 181, in main
    cli.main()
  File "/usr/local/lib/python2.7/site-packages/click-4.0-py2.7.egg/click/core.py", line 644, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/site-packages/click-4.0-py2.7.egg/click/core.py", line 991, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/site-packages/click-4.0-py2.7.egg/click/core.py", line 991, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/site-packages/click-4.0-py2.7.egg/click/core.py", line 837, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/site-packages/click-4.0-py2.7.egg/click/core.py", line 464, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click-4.0-py2.7.egg/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click-4.0-py2.7.egg/click/core.py", line 464, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/SoftLayer-4.0.2-py2.7.egg/SoftLayer/CLI/server/detail.py", line 100, in cli
    tag_row.append(tag['tag']['name'])
KeyError: 'tag'

Feel free to report this error as it is likely a bug:
    https://github.com/softlayer/softlayer-python/issues
```

I tried it under

```
➜  ~  python -V
Python 2.7.10
```

and 

```
(softlayer)➜  ~  python -V
Python 3.4.3
```

**Comment by underscorephil** [comment]
Could you email me the ID of one of the servers it fails on? pjackson@softlayer.com

**Comment by mbubb** [comment]
Hello Phil

I sent a list of affected servers to you r email

yrs

Michael

**Comment by underscorephil** [comment]
@mbubb I am unable to recreate this in the 4.x version. Try upgrading to the latest in pip.

**Comment by mbubb** [comment]
my packages are at
click-4.0 prettytable-0.7.2 requests-2.7.0 six-1.9.0 softlayer-4.0.4

and I still see:

```
(softlayer)➜  ~  slcli --format=raw server detail pix-ams-8
An unexpected error has occured:
Traceback (most recent call last):
  File "/Users/mbubb/.virtualenv/softlayer/lib/python3.4/site-packages/SoftLayer-4.0.2-py3.4.egg/SoftLayer/CLI/core.py", line 181, in main
    cli.main()
  File "/Users/mbubb/.virtualenv/softlayer/lib/python3.4/site-packages/click/core.py", line 644, in main
    rv = self.invoke(ctx)
  File "/Users/mbubb/.virtualenv/softlayer/lib/python3.4/site-packages/click/core.py", line 991, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/mbubb/.virtualenv/softlayer/lib/python3.4/site-packages/click/core.py", line 991, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/mbubb/.virtualenv/softlayer/lib/python3.4/site-packages/click/core.py", line 837, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/mbubb/.virtualenv/softlayer/lib/python3.4/site-packages/click/core.py", line 464, in invoke
    return callback(*args, **kwargs)
  File "/Users/mbubb/.virtualenv/softlayer/lib/python3.4/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/Users/mbubb/.virtualenv/softlayer/lib/python3.4/site-packages/click/core.py", line 464, in invoke
    return callback(*args, **kwargs)
  File "/Users/mbubb/.virtualenv/softlayer/lib/python3.4/site-packages/SoftLayer-4.0.2-py3.4.egg/SoftLayer/CLI/server/detail.py", line 100, in cli
    tag_row.append(tag['tag']['name'])
KeyError: 'tag'

Feel free to report this error as it is likely a bug:
    https://github.com/softlayer/softlayer-python/issues
(softlayer)➜  ~  python
Python 3.4.3 (v3.4.3:9b73f1c3e601, Feb 23 2015, 02:52:03)
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>>
```

**Comment by mbubb** [comment]
same for 2.7 ...

**Comment by underscorephil** [comment]
Seems @sudorandom found a fix. Will pull it in as soon as the tests finish.

**Comment by underscorephil** [comment]
@mbubb the master branch should have a fix for this. Please test it out and let us know.

**Comment by mbubb** [comment]
Thank you - looks like it worked

**Comment by mhca** [comment]
Hi,

The bug is still reproduced.

The following error is displayed when empty tags are created in the server:
|=================================
>slcli server detail 123456

An unexpected error has occured:
Traceback (most recent call last):
  File "C:\Python34\lib\site-packages\SoftLayer\CLI\core.py", line 180, in main
    cli.main()
  File "C:\Python34\lib\site-packages\click\core.py", line 644, in main
    rv = self.invoke(ctx)
  File "C:\Python34\lib\site-packages\click\core.py", line 991, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Python34\lib\site-packages\click\core.py", line 991, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Python34\lib\site-packages\click\core.py", line 837, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Python34\lib\site-packages\click\core.py", line 464, in invoke
    return callback(*args, **kwargs)
  File "C:\Python34\lib\site-packages\click\decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "C:\Python34\lib\site-packages\click\core.py", line 464, in invoke
    return callback(*args, **kwargs)
  File "C:\Python34\lib\site-packages\SoftLayer\CLI\server\detail.py", line 100, in cli
    tag_row.append(tag['tag']['name'])
KeyError: 'tag'
|=================================

Please review it.
@underscorephil , I will email you in order to provide more details to reproduce the problem.

Thanks.

**Comment by sudorandom** [comment]
@mhca What version do you have? The latest (with the fix) is 4.1.1.

```bash
slcli --version
slcli (SoftLayer Command-line), version 4.1.1
```

Based on the traceback I can see that that's erroring is not what it is supposed to be if you had the latest version. If you used pip (recommend) to install you can upgrade with this command:

```bash
pip install -U softlayer
```

**Comment by mhca** [comment]
I was working with 4.0.2 version. I updated it and it works successfully. Thanks for your help.
